include/BasicTypes.hpp

```diff
@@ -381,38 +381,65 @@ class NiStringRef {
 	bool operator!=(const std::string& rhs) const { return !operator==(rhs); }
 };
 
-template<typename ValueType, typename SizeType = uint32_t>
-class NiVector {
+// Helper to reduce duplication
+template<typename ValueType, typename SizeType>
+class NiVectorBase {
```

Comment:
Can you add `NiVectorBase::clear()`?

Reply:
> Have you tested the changes to NiVector with several files to see if they still work? (vector of size 0 and >0, string vector, string ref vector).

I have tested several hundred meshes. They all are optimized correctly